config/config.go

```diff
@@ -191,6 +197,29 @@ func (srvTL *ServiceTelemetryLogs) validate() error {
 	return nil
 }
 
+// ServiceTelemetryMetrics exposes the common Telemetry configuration for one component.
+type ServiceTelemetryMetrics struct {
```

Comment:
This is 1:1 translation of the current flags which are very specific to Prometheus exporter. We should design a solution that allows us to change to Otel and support different types of exporters such as OTLP or Prometheus.

Reply:
That's correct, this was intended as a first step to enable elimination of the CLI flags.  I think that to the extent that we do not currently know what the new design is then this may be the best thing to do for now.  We could add another layer to the configuration structures to hold the Prometheus-specific bits, but without knowing how that fits into the OTel SDK configuration model (which doesn't yet exist and which will need to be guarded by feature gates for some period of time after it does) we're only guessing as to whether it will be viable in the long run.